### PR TITLE
Discard campaign values if they are provided as arrays

### LINKED ISSUE #20189: TypeErrors in UrlHelper
Opening an issue on advice from the forums:
https://forum.matomo.org/t/mediaanalytics-error-message/48642/2

## Expected Behavior
No errors process tracking requests with array form GET parameters.

## Current Behavior
We're seeing type errors. 

## Possible Solution
We are currently using a patch to work around the type error from `utm` get param arrays:
```diff
diff --git core/UrlHelper.php core/UrlHelper.php
index 1e51828977..1faf819c63 100644
--- core/UrlHelper.php
+++ core/UrlHelper.php
@@ -279,7 +279,7 @@ class UrlHelper
     {
         $nameToValue = self::getArrayFromQueryString($urlQuery);
 
-        if (isset($nameToValue[$parameter])) {
+        if (isset($nameToValue[$parameter]) && is_string($nameToValue[$parameter])) {
             return $nameToValue[$parameter];
         }
         return null;
```

Note that this probably breaks the tests around `UrlHelper::getParameterFromQueryString`:
https://github.com/matomo-org/matomo/blob/8d05108aec86f5c6779b17d18c2e9a7ee7f2f827/tests/PHPUnit/Unit/UrlHelperTest.php#L117-L122

But those tests enforce contradiction with the docblock return type!
https://github.com/matomo-org/matomo/blob/8d05108aec86f5c6779b17d18c2e9a7ee7f2f827/core/UrlHelper.php#L275

This method should always return a string (not an array) in order to meet the documented interface.


## Steps to Reproduce (for Bugs)

<details>

<summary>
Tracking hits that include `utm` GET params in array format (which are not correct, but error out the tracker), such as
</summary>

```
[2022-12-21 20:11:45] piwik.ERROR: Uncaught exception: TypeError: urldecode(): Argument #1 ($string) must be of type string, array given in /var/www/html/plugins/Referrers/Columns/Base.php:379 Stack trace: #0 /var/www/html/plugins/Referrers/Columns/Base.php(379): urldecode(Array) #1 
/var/www/html/plugins/Referrers/Columns/Base.php(417): Piwik\Plugins\Referrers\Columns\Base->detectCampaignFromString('utm_campaign[]=...') #2 /var/www/html/plugins/Referrers/Columns/Base.php(560): Piwik\Plugins\Referrers\Columns\Base->detectReferrerCampaignFromLandingUrl() #3 /var/w
ww/html/plugins/Referrers/Columns/Base.php(121): Piwik\Plugins\Referrers\Columns\Base->detectReferrerCampaign(Object(Piwik\Tracker\Request), Object(Piwik\Tracker\Visitor)) #4 /var/www/html/plugins/Referrers/Columns/Base.php(267): Piwik\Plugins\Referrers\Columns\Base->getReferrerInfor
mation('', 'https://www.ifi...', 1, Object(Piwik\Tracker\Request), Object(Piwik\Tracker\Visitor)) #5 /var/www/html/plugins/Referrers/Columns/Keyword.php(35): Piwik\Plugins\Referrers\Columns\Base->getReferrerInformationFromRequest(Object(Piwik\Tracker\Request), Object(Piwik\Tracker\Vi
sitor)) #6 /var/www/html/core/Tracker/Visit.php(514): Piwik\Plugins\Referrers\Columns\Keyword->onNewVisit(Object(Piwik\Tracker\Request), Object(Piwik\Tracker\Visitor), Object(Piwik\Tracker\ActionPageview)) #7 /var/www/html/core/Tracker/Visit.php(317): Piwik\Tracker\Visit->triggerHook
OnDimensions(Array, 'onNewVisit') #8 /var/www/html/core/Tracker/Visit.php(210): Piwik\Tracker\Visit->handleNewVisit(NULL) #9 /var/www/html/core/Tracker.php(172): Piwik\Tracker\Visit->handle() #10 /var/www/html/plugins/QueuedTracking/Queue/Processor/Handler.php(46): Piwik\Tracker->tra
ckRequest(Object(Piwik\Tracker\Request)) #11 /var/www/html/plugins/QueuedTracking/Queue/Processor.php(194): Piwik\Plugins\QueuedTracking\Queue\Processor\Handler->process(Object(Piwik\Tracker), Object(Piwik\Tracker\RequestSet)) #12 /var/www/html/plugins/QueuedTracking/Queue/Processor.
php(143): Piwik\Plugins\QueuedTracking\Queue\Processor->processRequestSets(Object(Piwik\Tracker), Array) #13 /var/www/html/plugins/QueuedTracking/Commands/Process.php(88): Piwik\Plugins\QueuedTracking\Queue\Processor->process() #14 /var/www/html/vendor/symfony/console/Symfony/Compone
nt/Console/Command/Command.php(257): Piwik\Plugins\QueuedTracking\Commands\Process->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #15 /var/www/html/vendor/symfony/console/Symfony/Component/Console/Application.php(87
4): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #16 /var/www/html/vendor/symfony/console/Symfony/Component/Console/Application.php(195): Symfony\Component\Console\Application
->doRunCommand(Object(Piwik\Plugins\QueuedTracking\Commands\Process), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #17 [internal function]: Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Inpu
t\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #18 /var/www/html/core/Console.php(135): call_user_func(Array, Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #19 /var/www/html/core/Access.php(670): Piwi
k\Console->Piwik\{closure}() #20 /var/www/html/core/Console.php(136): Piwik\Access::doAsSuperUser(Object(Closure)) #21 /var/www/html/core/Console.php(87): Piwik\Console->doRunImpl(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput
)) #22 /var/www/html/vendor/symfony/console/Symfony/Component/Console/Application.php(126): Piwik\Console->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #23 /var/www/html/console(32): Symfony\Component\Console\Applica
tion->run() #24 {main} {"exception":"[object] (TypeError(code: 0): urldecode(): Argument #1 ($string) must be of type string, array given at /var/www/html/plugins/Referrers/Columns/Base.php:379)","ignoreInScreenWriter":true} {"class":"Piwik\\ExceptionHandler","request_id":138}
Uncaught exception in /var/www/html/plugins/Referrers/Columns/Base.php line 379:                                                                                                                                                                                                            
urldecode(): Argument #1 ($string) must be of type string, array given                                                                        
#0 /var/www/html/plugins/Referrers/Columns/Base.php(379): urldecode(Array)                                                                                                                                                                                                                  
#1 /var/www/html/plugins/Referrers/Columns/Base.php(417): Piwik\Plugins\Referrers\Columns\Base->detectCampaignFromString('utm_campaign[]=...')                                                                                                                                              
#2 /var/www/html/plugins/Referrers/Columns/Base.php(560): Piwik\Plugins\Referrers\Columns\Base->detectReferrerCampaignFromLandingUrl()                                                                                                                                                      
#3 /var/www/html/plugins/Referrers/Columns/Base.php(121): Piwik\Plugins\Referrers\Columns\Base->detectReferrerCampaign(Object(Piwik\Tracker\Request), Object(Piwik\Tracker\Visitor))
#4 /var/www/html/plugins/Referrers/Columns/Base.php(267): Piwik\Plugins\Referrers\Columns\Base->getReferrerInformation('', 'https://www.ifi...', 1, Object(Piwik\Tracker\Request), Object(Piwik\Tracker\Visitor))
#5 /var/www/html/plugins/Referrers/Columns/Keyword.php(35): Piwik\Plugins\Referrers\Columns\Base->getReferrerInformationFromRequest(Object(Piwik\Tracker\Request), Object(Piwik\Tracker\Visitor))
#6 /var/www/html/core/Tracker/Visit.php(514): Piwik\Plugins\Referrers\Columns\Keyword->onNewVisit(Object(Piwik\Tracker\Request), Object(Piwik\Tracker\Visitor), Object(Piwik\Tracker\ActionPageview))
#7 /var/www/html/core/Tracker/Visit.php(317): Piwik\Tracker\Visit->triggerHookOnDimensions(Array, 'onNewVisit')                                                                                                                                                                             
#8 /var/www/html/core/Tracker/Visit.php(210): Piwik\Tracker\Visit->handleNewVisit(NULL)                                                                                                                                                                                                     
#9 /var/www/html/core/Tracker.php(172): Piwik\Tracker\Visit->handle()                                                                         
#10 /var/www/html/plugins/QueuedTracking/Queue/Processor/Handler.php(46): Piwik\Tracker->trackRequest(Object(Piwik\Tracker\Request))                                                                                                                                                        
#11 /var/www/html/plugins/QueuedTracking/Queue/Processor.php(194): Piwik\Plugins\QueuedTracking\Queue\Processor\Handler->process(Object(Piwik\Tracker), Object(Piwik\Tracker\RequestSet))
#12 /var/www/html/plugins/QueuedTracking/Queue/Processor.php(143): Piwik\Plugins\QueuedTracking\Queue\Processor->processRequestSets(Object(Piwik\Tracker), Array)
#13 /var/www/html/plugins/QueuedTracking/Commands/Process.php(88): Piwik\Plugins\QueuedTracking\Queue\Processor->process()                                                                                                                                                                  
#14 /var/www/html/vendor/symfony/console/Symfony/Component/Console/Command/Command.php(257): Piwik\Plugins\QueuedTracking\Commands\Process->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#15 /var/www/html/vendor/symfony/console/Symfony/Component/Console/Application.php(874): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#16 /var/www/html/vendor/symfony/console/Symfony/Component/Console/Application.php(195): Symfony\Component\Console\Application->doRunCommand(Object(Piwik\Plugins\QueuedTracking\Commands\Process), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Outp
ut\ConsoleOutput))
#17 [internal function]: Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#18 /var/www/html/core/Console.php(135): call_user_func(Array, Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#19 /var/www/html/core/Access.php(670): Piwik\Console->Piwik\{closure}()                                                                                                                                                                                                                    
#20 /var/www/html/core/Console.php(136): Piwik\Access::doAsSuperUser(Object(Closure))                                                                                                                                                                                                       
#21 /var/www/html/core/Console.php(87): Piwik\Console->doRunImpl(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#22 /var/www/html/vendor/symfony/console/Symfony/Component/Console/Application.php(126): Piwik\Console->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#23 /var/www/html/console(32): Symfony\Component\Console\Application->run()                                                                                                                                                                                                                 
#24 {main}                        
```

</details>

Fail to be processed. 

## Context
These are "bad" tracking hits, in the sense that `utm_campaign[]=...` are not technically correct, but we should not be failing hard on processing these hits. Specifically, these errors stall out processing while using the QueuedTracking plugin, and stall all processing of the queue, similar to https://github.com/matomo-org/plugin-QueuedTracking/issues/192
 
## Your Environment

* Matomo Version: 4.12.3
* PHP Version: 8.0.26
* Server Operating System: Docker (alpine) (running on Fedora)
* Additionally installed plugins:
  - CustomReports                    
  - MarketingCampaignsReporting      
  - CustomAlerts                     
  - LogViewer                        
  - InvalidateReports                
  - TasksTimetable                   
  - QueuedTracking                   
  - AbTesting                        
  - MediaAnalytics                   
  - FormAnalytics                    
  - Funnels                          
  - RollUpReporting                  
  - SearchEngineKeywordsPerformance  
  - MultiChannelConversionAttribution
  - HeatmapSessionRecording          
  - SEOWebVitals                     
  - UsersFlow                        
  - ActivityLog                      
  - WhiteLabel                       
  - Cohorts                          
  - AdvertisingConversionExport      
  - LoginSaml                        
  - WooCommerceAnalytics             
  - CustomVariables                  
  - GoogleAnalyticsImporter

### PR DESCRIPTION
**Pull request by sgiehl**
### Description:

If campaign parameters are provided as arrays in the URL, that might currently let the tracking fail.
As arrays should be unexpected in that case, I guess simply ignoring such values should be a suitable solution.

fixes #20189

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Review by bx80 — APPROVED** [review]
Did some breakpoint checks on the new test, the test passes without issues and now no exception is thrown :+1:

* [x] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [x] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [x] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [x] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [x] n/a [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [x] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [x] Test added [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [x] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [x] n/a [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [x] n/a [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [x] n/a Existing documentation updated if needed
